src/analysis/seminorm.lean

```diff
@@ -0,0 +1,212 @@
+/-
+Copyright (c) 2019 Jean Lo. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jean Lo
+-/
+
+import algebra.pointwise
+import analysis.normed_space.basic
+
+/-!
+# Seminorms and local convexity
+
+This file introduces the following notions, defined for a vector space
+over a normed field:
+
+- the subset properties of being `absorbent` and `balanced`,
+
+- a `seminorm`, a function to the reals that is positive-semidefinite,
+  absolutely homogeneous, and subadditive.
+
+We prove related properties.
+
+## TODO
+
+Define and show equivalence of two notions of local convexity
+for a t.v.s. over ℝ or ℂ: that it has a local base of balanced convex
```

Comment:
I would avoid the abbreviation `t.v.s.` here and below, as spelling things explicitly can only help readers.